packages/oca/src/legacy/resolver/oca.ts

```diff
@@ -203,93 +144,119 @@ export class OCABundle implements OCABundleType {
         return overlay
       }
     }
-    return (this.bundle as Bundle).overlays.find((item) => item.type === type.toString()) as T
+    return this.bundle.overlays.find((item) => item.type === type.toString()) as T
   }
 }
 
-export class OCABundleResolver implements OCABundleResolverType {
-  private bundles: Record<string, Bundle>
-  private options: OCABundleResolverOptions
+export class DefaultOCABundleResolver implements OCABundleResolverType {
+  protected bundles: Record<string, OverlayBundle> = {}
+  protected options: OCABundleResolverOptions
+
+  public constructor(bundlesData: Record<string, IOverlayBundleData> = {}, options?: OCABundleResolverOptions) {
+    for (const cid in bundlesData) {
+      try {
+        this.bundles[cid] = new OverlayBundle(cid, bundlesData[cid])
+      } catch (error) {
+        // might get an error trying to parse javascript's default value
+      }
+    }
 
-  public constructor(bundles: Record<string, Bundle> = {}, options?: OCABundleResolverOptions) {
-    this.bundles = bundles
     this.options = {
-      cardOverlayType: options?.cardOverlayType ?? CardOverlayType.CardLayout11,
+      brandingOverlayType: options?.brandingOverlayType ?? BrandingOverlayType.Branding01,
       language: options?.language ?? 'en',
     }
   }
 
-  public get cardOverlayType(): CardOverlayType {
-    return this.options.cardOverlayType ?? CardOverlayType.CardLayout11
+  public get cardOverlayType(): BrandingOverlayType {
+    return this.options.brandingOverlayType ?? BrandingOverlayType.Branding01
   }
 
   private getDefaultBundle(params: { language?: string; identifiers?: Identifiers; meta?: Meta }) {
     if (!params.language) {
       params.language = 'en'
     }
-    const metaOverlay: MetaOverlay = {
-      captureBase: '',
+
+    const metaOverlay: IMetaOverlayData = {
+      capture_base: '',
       type: OverlayType.Meta10,
       name: startCase(
         params.meta?.credName ??
           parseCredDefFromId(params.identifiers?.credentialDefinitionId, params.identifiers?.schemaId)
       ),
-      issuerName: params.meta?.alias || params.meta?.credConnectionId || 'Unknown Contact',
+      issuer: params.meta?.alias || params.meta?.credConnectionId || 'Unknown Contact',
       language: params.language ?? this.options?.language,
+      description: '',
+      credential_help_text: '',
+      credential_support_url: '',
+      issuer_description: '',
+      issuer_url: '',
     }
 
     let colorHash = 'default'
     if (metaOverlay?.name) {
       colorHash = metaOverlay.name
-    } else if (metaOverlay?.issuerName) {
-      colorHash = metaOverlay.issuerName
+    } else if (metaOverlay?.issuer) {
+      colorHash = metaOverlay.issuer
     }
 
-    const cardLayoutOverlay10: CardLayoutOverlay10 = {
-      captureBase: '',
-      type: OverlayType.CardLayout10,
-      backgroundColor: hashToRGBA(hashCode(colorHash)),
+    const brandingoOverlay01: ILegacyBrandingOverlayData = {
+      capture_base: '',
```

Comment:
Mixed casing may break the component. Wonder if we just keep it in camelCase for now and map over the values as needed

Reply:
So technically this wont match the interface. You may have to specify one internally for now.